### PR TITLE
[5.0] Work around a successful upgrade with silent errors

### PR DESCRIPTION
**Pull request by Fedik**
### Summary of Changes

This is the same as 
- https://github.com/joomla/joomla-cms/pull/41367

But for 5.0 branch.
I think upmerge will be complicated, so I have re-apply changes.

### Testing Instructions
If that https://github.com/joomla/joomla-cms/pull/41367 will be tested, then this one also should be good.

Note: The patch https://github.com/joomla/joomla-cms/pull/41367 should be applied on 4.x version, before test.

**Test normal**
Try run update from 4.4 => 5.0
The update should work as usual.

**Test with error**
Simulate an error, add an exception `throw new Exception('Test error');` somehwere here https://github.com/joomla/joomla-cms/blob/7eff1b78215743a1f4002c9729dcd17e2aa35d4d/administrator/components/com_admin/script.php#L2557-L2565
And do update.

**Actual:** The update finished with Success message. And nothing in logs.
**Expected:** The update finished with error message. The error is logged.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed
- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/language/en-GB/com_joomlaupdate.ini [hunk 1]
```diff
@@ -75,13 +75,18 @@ COM_JOOMLAUPDATE_UPDATE_LOG_FILE="File %s downloaded."
 COM_JOOMLAUPDATE_UPDATE_LOG_FINALISE="Finalising installation."
 COM_JOOMLAUPDATE_UPDATE_LOG_INSTALL="Starting installation of new version."
 COM_JOOMLAUPDATE_UPDATE_LOG_START="Update started by user %2$s (%1$s). Old version is %3$s."
+COM_JOOMLAUPDATE_UPDATE_LOG_UPLOAD="Uploading update file"
 COM_JOOMLAUPDATE_UPDATE_LOG_URL="Downloading update file from %s."
+COM_JOOMLAUPDATE_UPDATE_LOGGING_TEST_FAIL="Logging does not work. Make sure the log folder is writable, and try again. The logging test failed with message: %s"
```
Comment (brianteeman) [inline administrator/language/en-GB/com_joomlaupdate.ini:81]:
```suggestion
COM_JOOMLAUPDATE_UPDATE_LOGGING_TEST_FAIL="Logging did not work. Make sure the log folder is writable, and try again. The logging test failed with message: %s"
```

File: administrator/language/en-GB/com_joomlaupdate.ini [hunk 2]
```diff
@@ -75,13 +75,18 @@ COM_JOOMLAUPDATE_UPDATE_LOG_FILE="File %s downloaded."
...
 COM_JOOMLAUPDATE_UPDATING_HEAD="Update in progress"
 COM_JOOMLAUPDATE_UPDATING_INPROGRESS="Please wait; Joomla is being updated. This may take a while."
 COM_JOOMLAUPDATE_UPDATING_FAIL="The update has failed."
 COM_JOOMLAUPDATE_UPDATING_COMPLETE="The update is finalising."
 COM_JOOMLAUPDATE_VIEW_COMPLETE_HEADING="Joomla Version Update Status"
+COM_JOOMLAUPDATE_VIEW_COMPLETE_INSTALLER_MESSAGE="Messages captured by installer"
 COM_JOOMLAUPDATE_VIEW_COMPLETE_MESSAGE="Your site has been updated. Your Joomla version is now %s."
+COM_JOOMLAUPDATE_VIEW_COMPLETE_UPDATE_ERRORS="Brief list of captured errors"
+COM_JOOMLAUPDATE_VIEW_COMPLETE_WITH_ERROR_MESSAGE="The update completed with errors. Please examine the update logs %s for more detail.<br>It is recommended to restore the site from backup, fix the issues that caused the update failure and try to update again.<br>You always can ask for help on Joomla! forum %s"
```
Comment (brianteeman) [inline administrator/language/en-GB/com_joomlaupdate.ini]:
```suggestion
COM_JOOMLAUPDATE_VIEW_COMPLETE_WITH_ERROR_MESSAGE="The update completed with errors. Please examine the update logs %s for more details.<br>It is recommended to restore the site from backup, fix the issues that caused the update failure and try to update again.<br>You always can ask for help on Joomla! forum %s"
```

File: administrator/language/en-GB/com_joomlaupdate.ini (hunk 1 above)
Comment (Fedik) [reply administrator/language/en-GB/com_joomlaupdate.ini:81]: Hm, this kind of message before update. Not really "a past", but current.
Like "Logging do not work currently, please check it before continue", maybe better `do`?

Comment (richard67) [reply administrator/language/en-GB/com_joomlaupdate.ini:81]: For singular and present tense, "does" is absolutely correct.

File: administrator/language/en-GB/com_joomlaupdate.ini (hunk 2 above)
Comment (richard67) [reply administrator/language/en-GB/com_joomlaupdate.ini]: Shouldn't this change also be made in the 4.3-dev PR?

File: administrator/language/en-GB/com_joomlaupdate.ini (hunk 1 above)
Comment (Fedik) [reply administrator/language/en-GB/com_joomlaupdate.ini:81]: Okay, I have restored it back to `does`

**Comment by richard67** [comment]
@Fedik I've allowed myself to trigger a branch update so I have the update package created by Drone consistent with current 5.0-dev. I will then create modified packages for testing like I did for the other 4.3-dev PR #41367 . I hope I can do it with less tests here than for the other one.

**Comment by HLeithner** [comment]
can I have some tests here please?

**Comment by richard67** [comment]
> can I have some tests here please?

@HLeithner Until when?

**Comment by Fedik** [comment]
@HLeithner the other one PR for 4.3 is tested https://github.com/joomla/joomla-cms/pull/41367, and this one is a copy, 
Merge this one, and it will tested all the way untill release :smile:

**Comment by HLeithner** [comment]
since the other pr is only a copy upmerge would make more sense...

**Comment by richard67** [comment]
> since the other pr is only a copy upmerge would make more sense...

@HLeithner That would be not easy because there have been made many changes in script.php for updates. Methods have been removed, and new methods have been added.

**Comment by HLeithner** [comment]
can you resolve the merge conflict please

**Comment by richard67** [comment]
@Fedik Shall I solve the conflicts for you?

**Comment by Fedik** [comment]
I will fix it,
@HLeithner  Did you added log for COM_JOOMLAUPDATE_UPDATE_LOG_UNINSTALL_EXTENSIONS?
It missing a real text string

**Comment by HLeithner** [comment]
Yes I added it and hope you fix it ;-)

**Comment by HLeithner** [comment]
thanks

**Pull request merged**